### ISSUE TITLE
EscalationFired — threshold-driven cognitive escalation event with uncertainty payload

### ISSUE THREAD

**Issue by wow-miley** [socket-link/ampere#502]
# Context

The Wave 3 cognitive event audit (socket-link/ampere#497) found no event type fires when uncertainty crosses an escalation threshold. The closest existing analogues are:

* `MessageEvent.EscalationRequested` (`MessageEvent.kt:86-107`) — a discretionary human-in-the-loop event with `reason`, `context`, and fixed `urgency = HIGH`. No uncertainty payload; not threshold-driven.
* `Event.QuestionRaised` (`Event.kt:74-102`) — agent-driven question with `questionText` and `context`. Also not threshold-driven.

Neither carries uncertainty value, threshold, or any signal that distinguishes "this fired because confidence dropped below threshold" from "the agent decided to ask."

The Lumos bridge (socket-link/ampere#498) needs to fire UNCERTAIN atmosphere + QUESTION glyph specifically when uncertainty escalation triggers. Mapping it to existing events would conflate threshold-driven uncertainty with discretionary escalation, producing visual noise on every human escalation regardless of confidence.

This ticket adds `CognitiveEvent.EscalationFired` — a first-class threshold-driven event with the uncertainty payload the bridge needs to drive correct visual behavior.

**On naming:** the prior thread's handoff used the term "CHI" (Cascading Hierarchical Inference / Confidence Threshold / similar). This codebase already speaks "Escalation" via `EscalationRequested`, `EscalationEventHandler`, `EscalationChannel.kt`, `AgentPause.kt`. To stay consistent, the new event is `EscalationFired`, not `ChiFired`. The two concepts are related but distinct: `EscalationRequested` is discretionary (agent decides to escalate), `EscalationFired` is threshold-driven (uncertainty crossed a configured threshold). Both exist; consumers can subscribe to either or both.

# Objective

Publish `CognitiveEvent.EscalationFired` on `EventSerialBus` whenever an agent's uncertainty value crosses the escalation threshold.

# Expected Outcomes

* New `@Serializable sealed interface CognitiveEvent : Event` in `ampere-core/.../event/CognitiveEvent.kt` (note: `CognitivePhaseEvent` from socket-link/ampere#500 is sibling, also `: Event`; they share the directory but not a hierarchy).
* `CognitiveEvent.EscalationFired` member:

  ```kotlin
  data class EscalationFired(
      val agentId: AgentId,
      val uncertaintyValue: Double,    // 0.0..1.0, where 1.0 is maximum uncertainty
      val threshold: Double,           // 0.0..1.0, the configured trip point
      val prompt: String,              // The question or context that triggered the evaluation
      val cognitivePhase: CognitivePhase?,  // Which phase was the agent in when the threshold was crossed
      override val eventSource: AgentId,
      override val urgency: EventUrgency = EventUrgency.HIGH,
      override val eventId: String = generateEventId(),
      override val timestamp: Long = currentTimeMillis(),
  ) : CognitiveEvent {
      companion object {
          const val EVENT_TYPE = "EscalationFired"
      }
  }
  ```
* The event is published from wherever the escalation-decision logic lives today. The audit pointed at `ampere-core/.../pause/EscalationChannel.kt` and `pause/AgentPause.kt` as candidate sites. Recon needed before implementation: find the call site where uncertainty is currently evaluated against any threshold. If no such evaluation exists, this ticket also adds it — see Technical Constraints below.
* `EscalationRequested` (the existing discretionary event) is not modified. The two events coexist.
* Test coverage:
  * Triggering an evaluation with `uncertaintyValue >= threshold` publishes `EscalationFired` with correct payload.
  * Triggering with `uncertaintyValue < threshold` does NOT publish `EscalationFired`. (Sibling AMPR-174 / Gap-3 covers the "considered but not fired" case as a separate event.)
  * Multiple rapid fires publish multiple events — no deduplication. (The bridge handles glyph queueing; the bus stays dumb.)
  * The event reaches subscribers through normal `EventSerialBus` dispatch.

# Technical Constraints

* The bridge's CHI/Escalation override design (in socket-link/ampere#498) requires this event to fire **at the moment uncertainty crosses the threshold**, not in lockstep with a phase transition or an LLM call. The publish site must be the evaluator itself.
* If AMPERE does not currently evaluate uncertainty against a configurable threshold (the recon may show this is missing), the implementer files a follow-up ticket for the evaluator and stubs out the publish call from a callable internal API. The bridge ticket can proceed once the call site exists, even if the evaluator implementation is minimal.
* The `uncertaintyValue` must be in `[0.0, 1.0]`. The implementer chooses the source (LLM logit probabilities, custom scoring, etc.) — file a sibling design ticket if the right uncertainty source is unclear.
* The `prompt` field carries human-readable context for the escalation. It is the same string the agent would surface to a human if the escalation flowed to a HITL system; for the bridge, it's metadata for telemetry / future tooling. The bridge does not display `prompt` in the orb (the orb has no text rendering at this layer).
* This event participates in normal `EventSerialBus` dispatch — concurrent handler execution, swallowed handler errors, etc. No special prioritization at the bus level.

# Tasks

1. Recon: find the actual escalation-decision site in `ampere-core/.../pause/`. Identify where uncertainty is (or could be) evaluated against a threshold. If the evaluation does not exist today, document the gap in this ticket's comments and decide whether to add a minimal evaluator here or file a follow-up. **Validate:** PR description cites the publish site by `file:line` or links to the follow-up ticket.
2. Define `CognitiveEvent` sealed interface and `EscalationFired` data class. Match payload structure to existing event types. **Validate:** `./gradlew :ampere-core:compileKotlin` succeeds.
3. Publish `EscalationFired` from the identified site whenever `uncertaintyValue >= threshold`. **Validate:** unit test triggers an evaluation above threshold, asserts the event arrives at a test subscriber within \~100ms with correct payload.
4. Add a no-publish test: `uncertaintyValue < threshold` produces no `EscalationFired` event. **Validate:** test as described.
5. Confirm `EscalationRequested` continues to fire on its existing discretionary path (touch nothing on that code path; just write a regression test that runs an existing escalation flow and asserts the event still arrives). **Validate:** test as described.
6. Document `EscalationFired` in the AMPERE event API docs / module README. Include a worked example of subscribing and a note distinguishing it from `EscalationRequested`. **Validate:** docs committed alongside code.

# Out of Scope

* `EscalationConsidered` event (the "considered but not fired" sibling). Filed separately as AMPR-174 / Gap-3.
* The bridge subscriber side. socket-link/ampere#498 owns subscribing to `EscalationFired`.
* The uncertainty source (LLM logits vs custom scoring). If the source is unclear, file a follow-up; this ticket can use a minimal placeholder evaluator.
* Renaming or deprecating `EscalationRequested`. Both events coexist.
* Threshold configuration UI / runtime tuning. Threshold lives wherever AMPERE config lives today.

# Reference

Audit document: [AMPERE Cognitive Event Surface (Wave 3 Audit)](<#>) — see Gap-2 section.
